src/components/application_manager/test/commands/hmi/ui_is_ready_request_test.cc

```diff
@@ -0,0 +1,209 @@
+    ON_CALL(app_mngr_, hmi_capabilities())
+        .WillByDefault(ReturnRef(mock_hmi_capabilities_));
+    Mock::VerifyAndClearExpectations(&mock_message_helper_);
+  }
+  void SetUpExpectations(bool is_ui_cooperating_available,
+                         bool is_send_message_to_hmi,
+                         bool is_message_contain_param,
+                         am::HmiInterfaces::InterfaceState state) {
+    EXPECT_CALL(mock_hmi_capabilities_,
+                set_is_ui_cooperating(is_ui_cooperating_available));
+
+    if (is_message_contain_param) {
+      EXPECT_CALL(app_mngr_, hmi_interfaces())
+          .WillRepeatedly(ReturnRef(mock_hmi_interfaces_));
+      EXPECT_CALL(
+          mock_hmi_interfaces_,
+          SetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_UI, state));
+    } else {
+      EXPECT_CALL(app_mngr_, hmi_interfaces())
+          .WillOnce(ReturnRef(mock_hmi_interfaces_));
+      EXPECT_CALL(mock_hmi_interfaces_, SetInterfaceState(_, _)).Times(0);
+    }
+
+    EXPECT_CALL(mock_hmi_interfaces_,
+                GetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_UI))
+        .WillOnce(Return(state));
+
+    if (is_send_message_to_hmi) {
+      ExpectSendMessagesToHMI();
+    }
+  }
+
+  void ExpectSendMessagesToHMI() {
+    smart_objects::SmartObjectSPtr get_language(CreateMessage());
+    (*get_language)[am::strings::params][am::strings::message_type] =
+        static_cast<int>(am::MessageType::kRequest);
+    (*get_language)[am::strings::msg_params] =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    EXPECT_CALL(mock_message_helper_,
+                CreateModuleInfoSO(hmi_apis::FunctionID::UI_GetLanguage, _))
+        .WillOnce(Return(get_language));
+
+    smart_objects::SmartObjectSPtr get_all_language(CreateMessage());
+    (*get_all_language)[am::strings::params][am::strings::message_type] =
+        static_cast<int>(am::MessageType::kRequest);
+    (*get_all_language)[am::strings::msg_params] =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    EXPECT_CALL(
+        mock_message_helper_,
+        CreateModuleInfoSO(hmi_apis::FunctionID::UI_GetSupportedLanguages, _))
+        .WillOnce(Return(get_all_language));
+
+    smart_objects::SmartObjectSPtr get_capabilities(CreateMessage());
+    (*get_capabilities)[am::strings::params][am::strings::message_type] =
+        static_cast<int>(am::MessageType::kRequest);
+    (*get_capabilities)[am::strings::msg_params] =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    EXPECT_CALL(mock_message_helper_,
+                CreateModuleInfoSO(hmi_apis::FunctionID::UI_GetCapabilities, _))
+        .WillOnce(Return(get_capabilities));
+
+    EXPECT_CALL(mock_hmi_capabilities_, set_handle_response_for(*get_language));
+    EXPECT_CALL(app_mngr_, ManageHMICommand(get_language));
+    EXPECT_CALL(app_mngr_, ManageHMICommand(get_all_language));
+    EXPECT_CALL(app_mngr_, ManageHMICommand(get_capabilities));
```

Comment:
@okozlovlux, do you really only expect that the same objects will be sent as argument, and don't interested in their values?